cli/azd/pkg/tools/kubectl/kubectl.go

```diff
@@ -0,0 +1,257 @@
+package kubectl
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/exec"
+	"github.com/azure/azure-dev/cli/azd/pkg/tools"
+	"github.com/drone/envsubst"
+)
+
+type KubectlCli interface {
```

Comment:
Did we decide to wrap kubectl instead of use k8s go sdk?

Reply:
Just starting with wrapping `kubectl` for this POC and will investigate option to use the k8s go SDK next.

Reply:
I think this is decision for long haul, so I'd like to consider tradeoffs before we ship this

Reply:
If your a k8s dev, you will have `kubectl` installed.  Long term yes, SDK would be preferred, but would like to ship first version without SDK and do this after.

Reply:
Please create a tracking issue.

Reply:
#1621